codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -69,20 +61,30 @@
  * <p>Outbound streams are supported via the {@link Http2StreamChannelBootstrap}.
  *
  * <p>{@link ChannelConfig#setMaxMessagesPerRead(int)} and {@link ChannelConfig#setAutoRead(boolean)} are supported.
+ *
+ * <h3>Reference Counting</h3>
+ *
+ * Some {@link Http2StreamFrame}s implement the {@link ReferenceCounted} interface, as they carry
+ * reference counted objects (e.g. {@link ByteBuf}s). The multiplex codec will call {@link ReferenceCounted#retain()}
+ * before propagating a reference counted object through the pipeline, and thus an application handler needs to release
+ * such an object after having consumed it. For more information on reference counting take a look at
+ * http://netty.io/wiki/reference-counted-objects.html
  */
 @UnstableApi
-public final class Http2MultiplexCodec extends ChannelDuplexHandler {
+public class Http2MultiplexCodec extends Http2ChannelDuplexHandler {
 
-    private static final InternalLogger logger = InternalLoggerFactory.getInstance(Http2MultiplexCodec.class);
+    private static final InternalLogger LOG = InternalLoggerFactory.getInstance(Http2MultiplexCodec.class);
 
-    private final Http2StreamChannelBootstrap bootstrap;
+    // Visible for testing
+    final Http2StreamChannelBootstrap bootstrap;
 
     private final List<Http2StreamChannel> channelsToFireChildReadComplete = new ArrayList<Http2StreamChannel>();
     private final boolean server;
-    private ChannelHandlerContext ctx;
+    // Visible for testing
+    ChannelHandlerContext ctx;
     private volatile Runnable flushTask;
 
-    private final IntObjectMap<Http2StreamChannel> childChannels = new IntObjectHashMap<Http2StreamChannel>();
```

Comment:
TODO: At some point after this PR was sent out, I had discussed with Jakob that I think it'd be best if each handler maintains its own map/list/set of streams. This was mainly is regard to per-stream properties (which was simplified here to `managedState()`, but I doubt it wouldn't become properties in the future), but it does impact several other things like `Http2GoAwayFrame`. I still think that would be best (and we can discuss it now or later if y'all disagree), although that means the stream lifecycle must be made very clear and _may not change_ after we go stable.

But that would mean removing the `childChannels` map and relying on the forEachActiveStream() would be a step in the wrong direction. That said, that change is a relatively small compared to many other changes here, and we could fix it later.